runtime/src/builtins.rs

```diff
@@ -12,25 +12,32 @@ pub struct Builtin {
     pub name: String,
     pub id: Pubkey,
     pub process_instruction_with_context: ProcessInstructionWithContext,
+    pub default_cost: u64,
```

Comment:
nit: pardon my ignorance with the overall intention around the change... how is this meant to evolve exactly? `default_cost` implies that it's a fallback in case nothing else is provided -- does this mean that sometimes the default cost is overridden? Or is it meant to be a base cost that can have other costs associated later on? If that's the case, `base_cost` may be a better term, or `base_compute_unit_cost` to be totally clear. What do you think?

Reply:
Happy to discuss this so I can learn more 😃 
 
I understand that some of the builtin programs have rather fixed execution path therefore static CU cost, but there are exceptions, for example, upgradable_loader can consume different amount of CUs depends on what it does exactly (re/deploy, upgrade etc). So ideally those more complicated programs can/will deducts actual consumed CUs just like bpf does. But before that happen, a `default` cost can be useful. 

I am thinking something like this in the follow up PR to make natives to consume CUs: 
```
fn process_executable_chain(...) {
    ...
    let pre_remaining_units = self.get_remaining();
    let result = (entry.process_instruction)(self);
    let post_remaining_units = self.get_remaining();
    *compute_units_consumed = pre_remaining_units.saturating_sub(post_remaining_units);

    // if builtin didn't deduct CUs during execution, then consume its default_compute_unit_cost
    // and check meter;
    if is_builtin && compute_units_consumed == 0 {
        self.consume_checked(builtin.default_compute_unit_cost)?;
    }
    ...
}
```

oh yeah, will call it `default_compute_unit_cost` to be clear.

Reply:
Ah ok, yeah `default_compute_unit_cost` sounds perfect! Thanks for the explanation